intents/Utilities/aspects.py

```diff
@@ -0,0 +1,185 @@
+            table = table.drop([row])
+
+    #some indices get deleted after slicing
+    table = table.reset_index()
+    # droping the new columnn named 'index' created after reset_index call
+    table = table.drop(['index'], axis=1)
+    return table
+
+def corp_other_columns(table, required_columns):
+    """This function removes the columns that are not in required_columns list
+       This would help getting rid of columns that are not to be displayed.
+    Args:
+        table: Type-pandas.dataframe
+        required_columns: Type-list of str
+
+    Returns:
+        Returns the table as a dataframe obj after removing the rest of the columns.
+    """
+
+    current_columns = table.columns
+    for column in current_columns:
+        if column not in required_columns:
+            table = table.drop([column], axis=1)
+
+    return table
+
+def count_distinct(values):
+    """ This function returns the number of distinct entries in values.
+    The values are inserted into a set and it's size is returned.
+
+    Args:
+        values: Type - anything that can be iterated over
+
+    Returns:
+        Count of distinct values.
+    """
+    distinct_set = set()
+    for value in values:
+        distinct_set.add(value)
+    return len(distinct_set)
+
+def group_by(table, dimensions, summary_operator):
+    """Groups the column by the columns in dimensions
+    Basically makes a map in which keys contain dimensions and values
+    are evaluated after applying the summary operator.
+
+    Args:
+        table: Type-pandas.dataframe
+            It has the contents of the csv file
+        dimensions: Type-list
+            It contains the names of columns according to which we group
+        summary_operator: Type-SummaryOperators enum members
+            It denotes the summary operator
+
+    Returns:
+        Returns the table as a dataframe obj after applying grouping
+    """
+    if summary_operator is None:
+        return table
+    if summary_operator == SummaryOperators.summation:
+        table = table.groupby(dimensions).sum()
+
+    if summary_operator == SummaryOperators.mean:
+        table = table.groupby(dimensions).mean()
+
+    if summary_operator == SummaryOperators.count:
+        table = table.groupby(dimensions).count()
+
+    if summary_operator == SummaryOperators.maximum:
+        table = table.groupby(dimensions).max()
+
+    if summary_operator == SummaryOperators.minimum:
+        table = table.groupby(dimensions).min()
+
+    if summary_operator == SummaryOperators.std:
+        table = table.groupby(dimensions).std()
+
+    if summary_operator == SummaryOperators.var:
+        table = table.groupby(dimensions).var()
+
+    if summary_operator == SummaryOperators.first:
+        table = table.groupby(dimensions).first()
+
+    if summary_operator == SummaryOperators.last:
+        table = table.groupby(dimensions).last()
+
+    if summary_operator == SummaryOperators.distinct:
```

Comment:
What if the given summary operator doesnt match any of the above? Should we have a default case?

Reply:
If we group the table by a list of columns, we would have to compress the rows using some summary operator. So in normal execution it has to match one of the if statements.
The right thing according is, convert the if statements into if-elif-else and in the else we raise an exception.